src/common/src/array/list_array.rs

```diff
@@ -160,6 +162,14 @@ pub struct ListArray {
     pub(super) value_type: DataType,
 }
 
+impl EstimateSize for ListArray {
+    fn estimated_heap_size(&self) -> usize {
+        self.bitmap.estimated_heap_size()
+            + self.offsets.capacity() * size_of::<u32>()
+            + self.value.estimated_heap_size()
```

Comment:
* Given `value: Box<ArrayImpl>`, it seems to me this should be `self.value.estimated_size()` to include `Self`?
* `DataType` could be recursive and also takes some space on the heap. What's the impact of underestimation?

Reply:
Yes, but I want to keep things simple for now since our goal is to count the most memory consume part, so leaving small things like `Arc`, `Box`, `DataType` is appropriate.